base/strings/io.jl

```diff
@@ -107,19 +107,26 @@ function sprint(f::Function, args...; context=nothing, sizehint::Integer=0)
     String(resize!(s.data, s.size))
 end
 
-tostr_sizehint(x) = 8
-tostr_sizehint(x::AbstractString) = lastindex(x)
-tostr_sizehint(x::Union{String,SubString{String}}) = sizeof(x)
-tostr_sizehint(x::Float64) = 20
-tostr_sizehint(x::Float32) = 12
+# CategoricalArrays extends this
+function tostr_sizehint end
```

Comment:
So this function isn't used now anymore? It seems somewhat unfortunate to duplicate this code in two places below just to avoid getting that extension. Maybe just
```
let tostr_sizehint(x) = begin
if x isa Float64
            siz += 20
        elseif x isa Float32
            siz += 12
        elseif x isa String || x isa SubString{String}
            siz += sizeof(x)
        elseif x isa Char
            siz += ncodeunits(x)
        else
            siz += 8
        end
end
```
around the two function definitions?

```

Reply:
Is the point of this change just to ensure CategoricalArrays don't overload this? Just ask if that's what you want -- that's more reliable than keeping an empty definition hoping that I don't notice. :-p

Why is it a problem exactly? I understand that things like `Base.convert(::Type{T}, ::CategoricalValue{T})` are hard for the compiler, but `Base.tostr_sizehint(::CategoricalString)` looks like a much more innocuous definition to me. Note that I don't care so much about `tostr_sizehint` as I care about `convert` (it's just a small optimization) so I'm just going to remove it.